woocommerce/wc-admin
# Fix a bug where the store task list would remain after completion

Fixes #5138. #5138's description is not quite accurate, because the task list should disappear itself after completion rather than requiring manual dismissal. This fixes the issue with dismissal as well as the setup link issue.

There is an issue on Atomic sites where for some reason the option `woocommerce_task_list_complete` is set to `no` at some point during the workflow of setting up a site. This is problematic for the task list because of some flawed logic in the component. Because the value gets set to `no`, the component can never update itself to the correct value. The logic [here](https://github.com/woocommerce/woocommerce-admin/blob/main/client/task-list/index.js#L73) always exits because `no` is a truthy value and the flawed logic would pass `no` as a prop value.

Everywhere else we coerce values like `no` and `yes` to a boolean before passing them to the component, so this matches that approach. Setting it to be a proper boolean means that it will correctly get updated when all the tasks a user has are completed (this check is done every time the component mounts or updates.)

### Detailed test instructions:

1. Create an Atomic site
2. Install latest WooCommerce
3. `npm run test:zip` on this branch and upload/install the zip file to your Atomic site's plugins
4. Activate both WooCommerce and WooCommerce admin
5. Complete the setup wizard
6. Complete all the tasks in the setup task list
7. Expected: The setup task list should vanish after your last task is complete.
8. Visit Customers link, the store setup link should no longer display in the activity panel.
### Changelog Note:

Fix: Bug where sometimes the task list would not vanish after being completed.
